core/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala

```diff
@@ -702,6 +702,13 @@ trait DeltaSQLConfBase {
         .checkValue(_ > 0, "'optimize.maxThreads' must be positive.")
         .createWithDefault(15)
 
+  val DELTA_OPTIMIZE_USE_REPARTITON =
+    buildConf("optimize.repartition.enabled")
+      .internal()
+      .doc("Use repartition(1) instead of coalesce(1) to merge small files.")
```

Comment:
This is used for both compacttion and zorder .. right? so this need not be small files.
Also, it would be good to elaborate when one should consider enabling this and the cost of enabling this (extra shuffle). For your benchmarks it seems that it is useful in the case (but not limited to) when you are compacting large number of small files.

Reply:
It's only for compaction job. https://github.com/delta-io/delta/blob/574d2ec30727afbbb25f54e6df4d4eebb608a5e8/core/src/main/scala/org/apache/spark/sql/delta/commands/OptimizeTableCommand.scala#L318

yes I'll update the description